xwiki-platform-core/xwiki-platform-mail/xwiki-platform-mail-send/xwiki-platform-mail-send-default/src/main/java/org/xwiki/mail/script/ScriptMimeMessage.java

```diff
@@ -90,6 +90,17 @@
      */
     public BodyPart addPart(String mimeType, Object content)
     {
+        if (content instanceof BodyPart) {
+            try {
+                Multipart multipart = getMultipart();
+                multipart.addBodyPart((BodyPart) content);
+            } catch (Exception e) {
+                // Save the exception for reporting through the script services's getError() API
+                setError(e);
+                return null;
+            }
+            return (BodyPart) content;
```

Comment:
Could you please rewrite it so that there's only 1 single return please? Otherwise looks good (maybe missing some javadoc to explain what objects are supported). Thanks

Reply:
Actually you put the "if" in the wrong method. You should put in the most specific method (the one with the most parameters) since otherwise calling that other method will not handle BodyPart input...

Reply:
@vmassol should I just put an if and then call the less specific method to handle it?

Reply:
You need to move the "If" to the other `addPart` method...

Reply:
to this one: `public BodyPart addPart(String mimeType, Object content, Map<String, Object> parameters)`

Reply:
okay, will do

Reply:
@akshitdewan as I indicated this is not about what do so when the content is already a BodyPart but what NOT to do when it's already a BodyPart. So the if here should be around the code that create a BodyPart when you don't already have one.